src/slide.js

```diff
@@ -34,87 +35,78 @@ import './slide.scss';
  */
 function Slide({ slide, id, run, slideDone, isNextSlide, prevSlideDuration }) {
   let slideComponent;
-
+  const slideExecution = new BaseSlideExecution(
```

Comment:
This should not be initialized here and slideDone should not be removed from ImageText, etc
This is to support different ways of signaling that the slide is done.